src/otx/core/model/module/base.py

```diff
@@ -103,9 +104,15 @@ def configure_optimizers(self) -> dict[str, Any]:
 
         :return: A dict containing the configured optimizers and learning-rate schedulers to be used for training.
         """
-        optimizer = self.hparams.optimizer(params=self.parameters())
+        if isinstance(self.hparams.optimizer, partial):
+            optimizer = self.hparams.optimizer(params=self.parameters())
+        else:
+            optimizer = self.hparams.optimizer
```

Comment:
Please change constructor's argument typing as well.
```suggestion
        optimizer = self.hparams.optimizer(params=self.parameters()) if isinstance(self.hparams.optimizer, OptimizerCallable) else self.hparams.optimizer
```
Change `scheduler` as well.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/2840/commits/3c24df17dccc6c610712a02b1d089cd409990837 - Thanks, I also fixed it to `if callable` instead of `if isinstance`.